core/src/main/java/org/apache/iceberg/ReachableFileUtil.java

```diff
@@ -45,7 +45,7 @@ private ReachableFileUtil() {}
    */
   public static String versionHintLocation(Table table) {
     // only Hadoop tables have a hint file and such tables have a fixed metadata layout
-    Path metadataPath = new Path(table.location(), METADATA_FOLDER_NAME);
+    Path metadataPath = new Path(table.location() + "/" + METADATA_FOLDER_NAME);
     Path versionHintPath = new Path(metadataPath, Util.VERSION_HINT_FILENAME);
```

Comment:
Why don't we need a similar +"/" here?

Reply:
When a slash is added, Path is able to understand there is a scheme followed by a path. In this case, it will add the `/` correctly. Anyway, I changed this line to use `this + "/" + that` syntax to avoid confusion.